protocols/generic/generic.go

```diff
@@ -106,34 +109,39 @@ func buildNetwork(tn *testnet.TestNet, nodeKeyPairs map[string]crypto.PrivKey, n
 
 		libp2p := tn.LDD.Params["libp2p"] == "true"
 
-		peerIds := map[int]string{}
-		for _, peerNode := range tn.Nodes {
-			if libp2p {
-				id, err := publicKeyToBase58(nodeKeyPairs[peerNode.GetID()])
-				if err != nil {
-					return util.LogError(err)
-				}
-				peerIds[peerNode.GetRelativeNumber()] = fmt.Sprintf(" --peers=/ip4/%s/tcp/%d/p2p/%s", peerNode.IP, p2pPort, id)
-			} else {
-				peerIds[peerNode.GetRelativeNumber()] = fmt.Sprintf(" --peers=%s", peerNode.IP)
-			}
+		if networkTopology != none {
 
-		}
+			params += fmt.Sprintf(" --port=%d", p2pPort)
 
-		peers, err := createPeers(node.GetRelativeNumber(), peerIds, networkTopology)
+			peerIds := map[int]string{}
+			for _, peerNode := range tn.Nodes {
+				if libp2p {
+					id, err := publicKeyToBase58(nodeKeyPairs[peerNode.GetID()])
+					if err != nil {
+						return util.LogError(err)
+					}
+					peerIds[peerNode.GetRelativeNumber()] = fmt.Sprintf(" --peers=/ip4/%s/tcp/%d/p2p/%s", peerNode.IP, p2pPort, id)
+				} else {
+					peerIds[peerNode.GetRelativeNumber()] = fmt.Sprintf(" --peers=%s", peerNode.IP)
```

Comment:
This should be GetAbsoluteNumber in case we build on multiple machines

Reply:
done